### PR TITLE
Add UILayoutSupport support

### PR DESCRIPTION
**Pull request by TimothyChilvers**
Hope you don't mind my taking a stab at this. It's based on the examples in https://github.com/robb/Cartography/issues/95 and admittedly it doesn't quite match the framework pattern, but it's here if you want it.

### CODE REVIEW

File: Cartography/Coefficients.swift [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
 public struct Coefficients {
-    var multiplier: CGFloat = 1
-    var constant: CGFloat = 0
+    public var multiplier: CGFloat = 1
+    public var constant: CGFloat = 0
 
-    init() { }
+    public init() { }
 
-    init(_ multiplier: CGFloat, _ constant: CGFloat) {
```
Comment (robb) [inline Cartography/Coefficients.swift]: Did you make this public for your testing or is there a reason for those changes?

File: Cartography/Expression.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ public struct Expression<T> {
     let value: T
     var coefficients: [Coefficients]
 
-    init(_ value: T, _ coefficients: [Coefficients]) {
```
Comment (robb) [inline Cartography/Expression.swift]: Same as above re making this public

File: Cartography/Property.swift [hunk 3]
```diff
@@ -144,7 +144,7 @@ public func >= <P: RelativeInequality>(lhs: P, rhs: Expression<P>) -> NSLayoutCo
     return lhs.context.addConstraint(lhs, coefficients: rhs.coefficients[0], to: rhs.value, relation: NSLayoutRelation.GreaterThanOrEqual)
 }
 
-// Mark: Addition
+//MARK: Addition
```
Comment (robb) [inline Cartography/Property.swift]: Nitpick: can you add a space here?

**Comment by robb** [comment]
Thanks for the pull request! 💖 

Would you mind adding some tests that also exercise the new API?

**Comment by CVertex** [comment]
Where's this at? Would love to see this in my favourite AutoLayout library :)

**Comment by TimothyChilvers** [comment]
I've been using it for months now and it's been fine, I'd like to add the tests and change the names to match styles that have been discussed elsewhere.

**Comment by CVertex** [comment]
great! Can't wait! :)

**Comment by orta** [comment]
@TimothyChilvers this looks good 👍  - it's worth trying to get this tested and merged

**Comment by TimothyChilvers** [comment]
All right, all right… I'll pick it up again now.

**Comment by TimothyChilvers** [comment]
Is my approach preferred over #212 ?

**Comment by orta** [comment]
@robb that opinion is on you I'm afraid ^

**Comment by TimothyChilvers** [comment]
I added tests, but I'm not happy with them. The reason being I can't make the UILayoutGuide of the view controller have a non-zero length just using UIKit components straight off. I only spent an hour or so fiddling with it, so I might have missed something. I have another branch which stubs the view controller and layout guide classes, but the compiler warnings for availability are a nightmare.

**Comment by TimothyChilvers** [comment]
So…

Having tried subclassing UIViewController and giving custom UILayoutGuide subclasses to the guides and failed, and failing to populate the UIKit supplied top and bottom layoutGuide objects with properties, and finally with UILayoutSupport not being equatable / internal behaviour and constraint generation being unreliable, I'm calling this as the limit of what I can achieve with the tests.

**Comment by orta** [comment]
Alright, given the comprehensive docs / tests on this PR, I'm gonna merge it tomorrow unless I hear a good to not

File: CartographyTests/LayoutSupportSpec.swift [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                
+                viewController.view.layoutIfNeeded()
+                
+                expect(view.convertRect(view.bounds, toView: window).maxY).to(equal((window.bounds.maxY - 100) - viewController.bottomLayoutGuide.length))
+            }
+            
+        }
+
+    }
+}
```
Comment (robb) [inline CartographyTests/LayoutSupportSpec.swift]: Nitpick: missing new line

Comment (TimothyChilvers) [inline CartographyTests/LayoutSupportSpec.swift]: I hate that too. Fixed.

**Comment by orta** [comment]
Hah - nice. OK, let's get this in.

**Pull request merged**